comparing a return value of c_inet_addr with a positive number.

kazu-yamamoto opened the pull request:
`c_inet_addr` returns a value of `Word32`.
So, we cannot compare it with (-1).
GHC 8.2 detects this bug.

vincenthz commented:
it's a bit sad this is raising a warning in this case (it's really not overflowing), but I would suggest fixing the warning not by hardcoding `-1`'s representation but instead using either:

```
+   if had == maxBound
+   if had == -(1)
```

The later `-(1)` trick shows how the warning is not consistent with the literal definition and haskell definition of negate/unary minus.

kazu-yamamoto commented:
@vincenthz Thank you for your suggestion. I took `maxBound`.

@eborden Than you for your review. I have modified, rebased and merged the commit.

I'm now trying to make history linear. So, I did `push -f` to this PR. And I will continue this procedure from now on since a linear history is really nice.
